### PR TITLE
Added server parsing logic

### PR DESCRIPTION
**Pull request by richardhuang09**
Searches user entry for "://" at the server login page.  If present, checks for "http" or "https" preceding "://".  If one of these is not found, an error is thrown.  If "://" is not found "https://" is prepended to the user entry.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@richardhuang09, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*
				
@richardhuang09, thanks for signing the CLA!

File: src/start/RealmScreen.js [hunk 1]
```diff
@@ -33,8 +33,22 @@ class RealmScreen extends React.Component {
...
+    let { realm } = this.state;
+    if (protocol.test(realm)) {
+      const tokens = realm.split(protocol);
+      switch (tokens[0]) {
+        case ('http'):
+          break;
+        case ('https'):
+          break;
+        default:
+          this.setState({ progress: false, error: 'Unsupported protocol' });
```
Comment (borisyankov) [inline src/start/RealmScreen.js]: You don't need to set progress to false here.
Not sure if explicitly handling the protocol as separate error is worth it.

**Comment by richardhuang09** [comment]
Neeraj wanted me to capture anything that looked like a protocol and then
return an error before calling the api if it wasn't http or https.  Are you
suggesting that we simply check for http or https and then append https to
the beginning of the user input if neither matches?

**Comment by borisyankov** [comment]
Yeah. That is what I meant.
It is very unlikely that a user will actually enter an incorrect protocol, while it is very likely that he will enter the url without a protocol.

**Comment by borisyankov** [comment]
Another, more general advise: the PR / commit messages are written in active voice (not always, but most projects nowadays, including this one). So instead of 'added' it would be 'add'.

**Comment by richardhuang09** [comment]
Thanks for the tips Boris.  I've change the code to reflect your comments.

**Comment by neerajwahi** [comment]
Thanks for the PR @richardhuang09 ! Merging now

**Pull request merged**